contracts/compliance/modular/modules/TimeTransfersLimitsModule.sol

```diff
@@ -0,0 +1,239 @@
+
+    /**
+     *  @dev See {IModule-moduleCheck}.
+     */
+    function moduleCheck(
+        address _from,
+        address /*_to*/,
+        uint256 _value,
+        address _compliance
+    ) external view override returns (bool) {
+        if (_from == address(0)) {
+            return true;
+        }
+
+        if (_isTokenAgent(_compliance, _from)) {
+            return true;
+        }
+
+        address senderIdentity = _getIdentity(_compliance, _from);
+        for (uint256 i = 0; i < transferLimits[_compliance].length; i++) {
+            if (_value > transferLimits[_compliance][i].limitValue) {
+                return false;
+            }
+
+            if (!_isUserCounterFinished(_compliance, senderIdentity, transferLimits[_compliance][i].limitTime)
+                && usersCounters[_compliance][senderIdentity][transferLimits[_compliance][i].limitTime].value + _value
+                    > transferLimits[_compliance][i].limitValue) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+    *  @dev getter for `transferLimits` variable
+    *  @param _compliance the Compliance smart contract to be checked
+    *  returns array of Limits
+    */
+    function getTimeTransferLimits(address _compliance) external view returns (Limit[] memory limits) {
+        return transferLimits[_compliance];
+    }
+
+    /**
+    *  @dev Checks if the cooldown must be reset, then increases user's OnchainID counters,
+    *  @param _compliance the Compliance smart contract address
+    *  @param _userAddress user wallet address
+    *  @param _value, value of transaction)to be increased
+    *  internal function, can be called only from the functions of the Compliance smart contract
+    */
+    function _increaseCounters(address _compliance, address _userAddress, uint256 _value) internal {
+        address identity = _getIdentity(_compliance, _userAddress);
+        for (uint256 i = 0; i < transferLimits[_compliance].length; i++) {
+            _resetUserCounter(_compliance, identity, transferLimits[_compliance][i].limitTime);
+            usersCounters[_compliance][identity][transferLimits[_compliance][i].limitTime].value += _value;
+        }
+    }
+
+    /**
+    *  @dev resets cooldown for the user if cooldown has reached the time limit
+    *  @param _compliance the Compliance smart contract address
+    *  @param _identity ONCHAINID of user wallet
+    *  @param _limitTime limit time frame
+    *  internal function, can be called only from the functions of the Compliance smart contract
+    */
+    function _resetUserCounter(address _compliance, address _identity, uint32 _limitTime) internal {
+        if (_isUserCounterFinished(_compliance, _identity, _limitTime)) {
+            usersCounters[_compliance][_identity][_limitTime].timer = block.timestamp + _limitTime;
+            usersCounters[_compliance][_identity][_limitTime].value = 0;
```

Comment:
Minor optimization here as well regarding mappings (not very useful there, but it's good to take the habit to write solidity like this).

```suggestion
            TransferCounter storage counter = usersCounters[_compliance][_identity][_limitTime];
            counter.timer = block.timestamp + _limitTime;
            counter.value = 0;
```

Reply:
Thank you for all these tips 🙏 done